### ISSUE TITLE
omni fails when tmp directory is on another drive

### ISSUE THREAD

**Issue by seledwards** [GENI-NSF/geni-tools#661]
When running `omni` on a computer (in this case a Linux machine) where the default `tmp` directory is on another drive, omni is unuseable.

What happens is that omni fails to download the `agg_nick_cache` AND then fails to proceed even if a AM URL is provided.  

Tomasz Buchert (cc'd) has provided a patch for this which is attached.  Please acknowledge him as a contributor when incorporating this patch.

 _Imported from trac ticket [#661](http://trac.gpolab.bbn.com/gcf/ticket/661),  created by sedwards on 07-07-2014 at 10:53, last modified: 07-07-2014 at 12:55_
   CCing: tomasz.buchert@..., gpo-expt-support@...

**Comment by seledwards** [comment]
_sedwards (seledwards) attached [0001-fix-a-bug-triggered-when-tmp-is-on-a-different-mount.patch](http://trac.gpolab.bbn.com/gcf/attachment/ticket/661/0001-fix-a-bug-triggered-when-tmp-is-on-a-different-mount.patch) on 07-07-2014 at 10:53_

**Comment by ahelsing** [comment]
Is this the same issue as #646?

_Trac comment by bujcich on 07-07-2014 at 11:06_

**Comment by innnkkki** [comment]
Yep it looks pretty similar, other than the fact that failure to download agg_nick_cache should not prevent omni from executing. 
E.g. our wiki being down is not a reason for omni not to run with an older cache file. 

Replying to [bujcich](comment:1):
> Is this the same issue as #646?


_Trac comment by nriga (github user: innnkkki) on 07-07-2014 at 11:16_

**Comment by seledwards** [comment]
I just created #662 to fix the issue with omni failing to run.  This is a regression bug.

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 11:21_

**Comment by seledwards** [comment]
Tomas adds:

```
Just to make sure that it is not misinterpreted:
the file succeeds to download, but fails to be *moved* (with os.rename)
because /tmp and my home directory are in different mounts.
```

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 12:00_

**Comment by seledwards** [comment]
I agree this ticket is a duplicate of #646. 

I've sent Tomasz the fix for #646 and #662 via email if he'd like to test those fixes.  

I'll reopen if he sees any remaining issues.

_Trac comment by sedwards (github user: seledwards) on 07-07-2014 at 12:55_
